src/System.Linq.Expressions/src/System/Dynamic/Utils/TypeExtensions.cs

```diff
@@ -17,9 +17,14 @@ internal static partial class TypeExtensions
             string name,
             Type[] types)
         {
-            var method = type.GetAnyStaticMethod(name);
-
-            return method.MatchesArgumentTypes(types) ? method : null;
+            foreach (MethodInfo method in type.GetRuntimeMethods())
```

Comment:
It seems this should be DeclaredMethods. 
We should be searching only in the given type for compat with .Net FX with does not pass FlattenHieararchy when searching for operators.
Possibly the original code had a bug here.

We need a test to validate the "no flattening" behavior